src/Handshake/ServerNegotiator.php

```diff
@@ -17,8 +17,24 @@ class ServerNegotiator implements NegotiatorInterface {
 
     private $_strictSubProtocols = false;
 
-    public function __construct(RequestVerifier $requestVerifier) {
+    private $enablePerMessageDeflate = false;
+
+    public function __construct(RequestVerifier $requestVerifier, $enablePerMessageDeflate = false) {
         $this->verifier = $requestVerifier;
+
+        // https://bugs.php.net/bug.php?id=73373
+        // https://bugs.php.net/bug.php?id=74240 - need >=7.1.4 or >=7.0.18
+        $supported = PermessageDeflateOptions::permessageDeflateSupported();
+        if ($enablePerMessageDeflate && !$supported) {
+            trigger_error('permessage-deflate is being disabled because it is not support by your PHP version.', E_USER_NOTICE);
```

Comment:
This may be buried deeper in some library that enables deflate by default. In that case an exception will just kill things without the end user being able to fix it. But I see your point too. Let me know which way you prefer and I can change if needed.

Reply:
@mbonneau In that case it's the duty of the library to make this option configurable. It's not the job of a library to mitigate nested library issues due to missing configuration options on the other library side.